### PR TITLE
Update to Gradle 7.5

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
 Update to Gradle 7.5
 
### Issues Resolved
Closes https://github.com/opensearch-project/job-scheduler/issues/207
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -7,7 +7,7 @@ import org.opensearch.gradle.test.RestIntegTestTask
 
 buildscript {
     ext {
-        opensearch_version = System.getProperty("opensearch.version", "2.1.0-SNAPSHOT")
+        opensearch_version = System.getProperty("opensearch.version", "2.2.0-SNAPSHOT")
```
Comment (dblock) [inline build.gradle:10]: Main should be on 3.0, no? Either way I would make this change separately.

**Review by dblock — DISMISSED** [review]

File: build.gradle (hunk 1 above)
Comment (dblock) [reply build.gradle:10]: https://github.com/opensearch-project/job-scheduler/pull/206/files

Comment (reta) [reply build.gradle:10]: Thanks @dblock , I think for most plugin `main` is kept in sync with next OpenSearch release (it seems like `job-scheduler` has not released `2.1.0.0`)

Comment (dblock) [reply build.gradle:10]: See https://github.com/opensearch-project/opensearch-plugins/issues/142, I think it's happening, cc: @prudhvigodithi

**Review by dblock — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Hey @reta I also noticed there is change in few lines with [gradle wrapper ](https://github.com/gradle/gradle/blob/v7.5.1/gradlew)`gradlew`  for 7.5.1, assuming this should be fine?
Thank you

**Review by VachaShah — APPROVED** [review]

**Review by saratvemulapalli — DISMISSED** [review]

**Comment by saratvemulapalli** [comment]
@reta thanks for the change. Looks like tests are failing, could you take a look?

**Comment by reta** [comment]
> @reta thanks for the change. Looks like tests are failing, could you take a look?

@saratvemulapalli those are flaky ones :(

```

Suite: Test class org.opensearch.jobscheduler.spi.utils.LockServiceIT
  1> [2022-08-31T20:38:34,310][INFO ][o.o.j.s.u.LockServiceIT  ] [testRenewLock] before test
  2> أغس 31, 2022 8:38:41 م com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
  2> WARNING: Uncaught exception in thread: Thread[opensearch[external_0][transport_worker][T#1],5,TGRP-LockServiceIT]
  2> java.lang.AssertionError: Failed to release lock.
  2> 	at __randomizedtesting.SeedInfo.seed([54A0208AAE2953EB]:0)
  2> 	at org.junit.Assert.fail(Assert.java:89)
  2> 	at org.junit.Assert.assertTrue(Assert.java:42)
  2> 	at org.opensearch.jobscheduler.spi.utils.LockServiceIT.lambda$testRenewLock$69(LockServiceIT.java:494)
  2> 	at org.opensearch.action.ActionListener$1.onResponse(ActionListener.java:80)
  2> 	at org.opensearch.jobscheduler.spi.utils.LockService.lambda$release$10(LockService.java:286)
  2> 	at org.opensearch.action.ActionListener$1.onResponse(ActionListener.java:80)
  2> 	at org.opensearch.jobscheduler.spi.utils.LockService.lambda$updateLock$5(LockService.java:208)
  2> 	at org.opensearch.action.ActionListener$1.onFailure(ActionListener.java:88)
  2> 	at org.opensearch.action.support.TransportAction$1.onFailure(TransportAction.java:12
```

Tracked as https://github.com/opensearch-project/job-scheduler/issues/56

File: build.gradle [hunk 2]
```diff
@@ -7,7 +7,7 @@ import org.opensearch.gradle.test.RestIntegTestTask
 
 buildscript {
     ext {
-        opensearch_version = System.getProperty("opensearch.version", "2.2.0-SNAPSHOT")
+        opensearch_version = System.getProperty("opensearch.version", "2.3.0-SNAPSHOT")
```
Comment (reta) [inline build.gradle:10]: Brings `snakeyml` update: https://github.com/opensearch-project/OpenSearch/blob/2.x/buildSrc/version.properties#L14

**Comment by reta** [comment]
The  WhiteSource Security Check should be fixed with new `2.3.0-SNAPSHOT`s (see please https://github.com/opensearch-project/job-scheduler/pull/208#discussion_r959897427).

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@saratvemulapalli care to ack? Let’s backport this to 2.x and increment main to 3.0 next (I’ll do it if I don’t see someone else on it)

**Comment by dblock** [comment]
@reta if you want to bump it to 3.0 on main now that’s cool with me

**Review by saratvemulapalli — APPROVED** [review]

**Comment by reta** [comment]
> @reta if you want to bump it to 3.0 on main now that’s cool with me

I will do that shortly, thanks @dblock !

**Pull request merged**